integration/feature/init/src/MillInitMavenTests.scala

```diff
@@ -0,0 +1,221 @@
+package mill.integration
+
+import mill.testkit.{IntegrationTester, UtestIntegrationTestSuite}
+import utest._
+
+object MillInitMavenTests extends UtestIntegrationTestSuite {
+
+  private def prep[T](githubSourceZipUrl: String)(f: IntegrationTester => T): T = {
+    val zipFile = os.temp(requests.get(githubSourceZipUrl))
+    val unzipDir = os.unzip(zipFile, os.temp.dir())
+    val sourceDir = os.list(unzipDir).head
+
+    integrationTest { tester =>
+      for (p <- os.list(sourceDir)) os.move.into(p, tester.workspacePath)
+      f(tester)
+    }
+  }
+
+  def tests: Tests = Tests {
+
+    // - Junit5
+    // - maven-compiler-plugin release option
+    test("jansi") {
+      prep("https://github.com/fusesource/jansi/archive/refs/tags/jansi-2.4.1.zip") { tester =>
+        import tester._
+
+        val initRes = eval(("init", "--deps-object", "Deps"))
+        initRes.out.contains(
+          "generated 1 Mill build file(s)"
+        ) ==> true
+        initRes.isSuccess ==> true
+
+        val compileRes = eval("compile")
+        compileRes.err.contains(
+          "compiling 20 Java sources"
+        ) ==> true
+        compileRes.isSuccess ==> true
+
+        val testRes = eval("test")
+        testRes.out.contains(
+          "Test run finished: 0 failed, 1 ignored, 90 total"
+        ) ==> true
+        testRes.isSuccess ==> true
+
+        val publishLocalRes = eval("publishLocal")
+        publishLocalRes.err.contains(
+          "Publishing Artifact(org.fusesource.jansi,jansi,2.4.1)"
+        ) ==> true
+        publishLocalRes.isSuccess ==> true
+      }
+    }
+
+    // - multi-module
+    // - TestNg
+    // - maven-compiler-plugin release option
+    test("dotenv") {
+      prep("https://github.com/shyiko/dotenv/archive/refs/tags/0.1.1.zip") { tester =>
+        import tester._
+
+        val initRes = eval("init")
+        initRes.out.contains(
+          "generated 3 Mill build file(s)"
+        ) ==> true
+        initRes.isSuccess ==> true
+
+        val resolveRes = eval(("resolve", "_"))
+        Seq(
+          "dotenv",
+          "dotenv-guice"
+        ).forall(resolveRes.out.contains) ==> true
+
+        // JavaModule.JavaTests is not picking compileIvyDeps from outer module
```

Comment:
This can be fixed by adding the following in `JavaModule.JavaTests`,
```scala
override def compileIvyDeps = outer.compileIvyDeps()
```
Is this the correct solution?